### PR TITLE
Add PlayerInsertLecternBookEvent

### LINKED ISSUE #2286: PlayerLecternPutEvent
**Is your feature request related to a problem? Please describe.**
Event that will be fired when player put book on Lectern.

**Describe the solution you'd like**
Add event that will be fired when player puts book on lectern.

**Describe alternatives you've considered**
Canceling right click will also block option to read book by players.

### PR DESCRIPTION
**Pull request by Proximyst**
Surprisingly big diff for such a small event...

Closes #2286.

### CODE REVIEW

File: Spigot-API-Patches/0222-Add-PlayerInsertLecternBookEvent.patch [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
++        return book;
++    }
++
++    @Override
++    public void setCancelled(boolean cancelled) {
++        this.cancelled = cancelled;
++    }
++
++    public void setBook(@Nullable ItemStack book) {
++        if (book == null)
```
Comment (kennytv) [inline Spigot-API-Patches/0222-Add-PlayerInsertLecternBookEvent.patch]: *borkets*

Comment (Proximyst) [reply Spigot-API-Patches/0222-Add-PlayerInsertLecternBookEvent.patch]: sorry mr wandering knenwald, your borkets aren't a valid currency here

Comment (kennytv) [reply Spigot-API-Patches/0222-Add-PlayerInsertLecternBookEvent.patch]: **brokcets**

File: Spigot-Server-Patches/0594-Add-PlayerInsertLecternBookEvent.patch [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+         this.page = 0;
+@@ -216,6 +218,13 @@ public class TileEntityLectern extends TileEntity implements Clearable, ITileInv
+     private ItemStack b(ItemStack itemstack, @Nullable EntityHuman entityhuman) {
+         if (this.world instanceof WorldServer && itemstack.getItem() == Items.WRITTEN_BOOK) {
+             ItemWrittenBook.a(itemstack, this.a(entityhuman), entityhuman);
++            // Paper start
++            if (entityhuman != null && getWorld() != null && !new PlayerInsertLecternBookEvent(
++                (org.bukkit.entity.Player) entityhuman.getBukkitEntity(),
++                (org.bukkit.block.Lectern) getWorld().getWorld().getBlockAt(position.getX(), position.getY(), position.getZ()).getState(false),
++                org.bukkit.craftbukkit.inventory.CraftItemStack.asCraftMirror(itemstack)
```
Comment (electronicboy) [inline Spigot-Server-Patches/0594-Add-PlayerInsertLecternBookEvent.patch:101]: looks like you never touch the new book if it's been changed in the event

**Review by electronicboy — CHANGES_REQUESTED** [review]

**Comment by Machine-Maker** [comment]
Superseded by https://github.com/PaperMC/Paper/pull/7305

**Pull request closed without merging**